parsec/core/gui/invite_user_widget.py

```diff
@@ -1,206 +1,276 @@
 # Parsec Cloud (https://parsec.cloud) Copyright (c) AGPLv3 2019 Scille SAS
 
-from PyQt5.QtCore import pyqtSignal, QPoint
-from PyQt5.QtWidgets import QToolTip, QWidget, QApplication
+from uuid import UUID
+
+from PyQt5.QtCore import pyqtSignal, Qt
+from PyQt5.QtGui import QFontMetrics
+from PyQt5.QtWidgets import QWidget, QApplication, QMenu
 
 from structlog import get_logger
 
-from parsec.api.protocol import UserID
-from parsec.core.invite_claim import (
-    InviteClaimTimeoutError,
-    InviteClaimBackendOfflineError,
-    InviteClaimError,
-    generate_invitation_token as core_generate_invitation_token,
-    invite_and_create_user as core_invite_and_create_user,
-)
-from parsec.core.types import BackendOrganizationClaimUserAddr
-
-from parsec.core.backend_connection import BackendNotAvailable, BackendConnectionError
+from parsec.api.protocol import InvitationType, InvitationStatus, InvitationDeletedReason
+from parsec.core.types import BackendInvitationAddr
+
+from parsec.core.backend_connection import backend_authenticated_cmds_factory
 from parsec.core.gui import desktop
-from parsec.core.gui import validators
-from parsec.core.gui.custom_dialogs import show_info, show_error, GreyedDialog
+from parsec.core.gui.greet_user_widget import GreetUserWidget
+from parsec.core.gui.custom_dialogs import show_error, GreyedDialog, ask_question
 from parsec.core.gui.lang import translate as _
+from parsec.core.gui.trio_thread import JobResultError, ThreadSafeQtSignal, QtToTrioJob
 from parsec.core.gui.ui.invite_user_widget import Ui_InviteUserWidget
-from parsec.core.gui.trio_thread import JobResultError, ThreadSafeQtSignal
+from parsec.core.gui.ui.user_invitation_widget import Ui_UserInvitationWidget
 
 
 logger = get_logger()
 
 
-async def _do_registration(core, device, new_user_id, token, is_admin):
-    try:
-        new_user_id = UserID(new_user_id)
-    except ValueError as exc:
-        raise JobResultError("registration-invite-bad-value") from exc
+async def _do_invite_user(device, config, email):
+    async with backend_authenticated_cmds_factory(
```

Comment:
Don't want to use `core.backend_cmds` instead ? this would have two small benefits: a) slightly better performances and b) if this fail due to poor connection, the whole application is considered offline